can-bus/common.c

```diff
@@ -0,0 +1,368 @@
+/* common.c
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include "common.h"
+
+int sock;
+volatile int keep_running = 1;
+
+static uint8_t *copy_buf = NULL;
+static uint8_t *copy_buf_ptr;
+static size_t copy_buf_len;
+static IsoTpLink g_link;
+
+/* Alloc send and receive buffer statically in RAM */
+static uint8_t g_isotpRecvBuf[ISOTP_BUFSIZE];
+static uint8_t g_isotpSendBuf[ISOTP_BUFSIZE];
+
+void sig_handle(int dummy);
+
+/* Debug function required by ISO-TP to compile, can be blank */
+void isotp_user_debug(const char* format, ...) {
+    va_list args;
+    va_start(args, format);
+    vprintf(format, args);
+    printf("\r\n");
+    va_end(args);
+}
+
+/* Timing function required by ISO-TP to compile */
+uint32_t isotp_user_get_ms(void) {
+    struct timeval t;
+    gettimeofday(&t, NULL);
+    return (t.tv_sec * 1000) + (t.tv_usec / 1000);
+}
+
+/* CAN bus send function require by ISO-TP to compile */
+int isotp_user_send_can(const uint32_t arbitration_id, const uint8_t* data,
+        const uint8_t size) {
+    /* Copy the message into a CAN bus frame, the message will always be
+     * 8 bytes or less */
+    struct can_frame frame;
+    frame.can_id = arbitration_id;
+    frame.can_dlc = size;
+    memcpy(frame.data, data, size);
+
+    /* Write the frame to the CAN bus */
+    if (write(sock, &frame, sizeof(struct can_frame))
+            != sizeof(struct can_frame)) {
+       perror("Write error\n");
+       return -1;
+    }
+
+    return 0;
+}
+
+/* Our CAN bus receive function */
+int can_receive(uint8_t data[CAN_MSG_LEN], int *length) {
+    int nbytes;
+    struct can_frame frame;
+    struct pollfd p[1];
+
+	p[0].fd = sock;
+	p[0].events = POLLIN;
+
+    /* Poll for new data */
+	int retval = poll(p, 1, 10);
```

Comment:
All variable decorations need to be at top of function or within brace.

Reply:
Adding `-Wdeclaration-after-statement` in the next commit to detect these in future.